sbatches/farmshare/singularity-jupyterlab.sbatch

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -v  
+
+# Usage
+
+# 1. Default Jupyter notebook (with your scratch to work in)
+# $ bash start.sh singularity-jupyter
+
+# 2. Default Jupyter notebook with custom working directory
+# $ bash start.sh singularity-jupyter /scratch/users/<username>
+
+# 3. Select your own jupyter container on Sherlock!
+# $ bash start.sh singularity-jupyter /scratch/users/<username> /path/to/container
+
+# 4. Or any singularity container...
+# $ bash start.sh singularity /path/to/container <args>
+
+PORT=$1
+NOTEBOOK_DIR=/farmshare/scratch/users/$USER
+#CONTAINER=/farmshare/home/classes/bioe/301p/ce/ce
+
+export SINGULARITY_CACHEDIR=/farmshare/user_data/${USERNAME}
+echo "Container is ${CONTAINER}"
+echo "Notebook directory is ${NOTEBOOK_DIR}"
+cd ${NOTEBOOK_DIR}
+
+#Create .local folder for default modules, if doesn't exist
+#if [ ! -d "${HOME}/.local" ];
+#  then
+#    echo "Creating local python modules folder to map at ${HOME}/.local";
+#    mkdir -p "${HOME}/.local";
+#fi
+
+. /etc/profile
+module load singularity/3.4.0
```

Comment:
How will they install modules without binding a local directory?

Reply:
I was thinking that they could create their own virtual environments to run Jupyter Kernel from and have modules installed in the virtual environment from the command line. We would run jupyter notebook from the virtual environment. There are really good tutorials on how to add the virtual environment to the Jupyter Notebook.

Possibly too advanced? or should I keep the line to create a local directory. Your call since you have more data on who actually uses this utility.

Reply:
The only issue is writable - if the containers are automatically binding home and the virtual environments can write there, then you are good. But if the container defaults to something like /home/joyvan then either you need to specify the user `--home` to be that, or bind the directory that needs to be written to explicitly (e.g., this is what I did with the .local folder int the other recipes).

Reply:
I am defaulting to your method and binding the local folder for farmshare (I think for stability it may just be better). Updated on b3df7dd